### PR TITLE
Video input texture and setup.

### PR DESCRIPTION
**Pull request by henryouly**
- Added `InputController` component.
- New Texture: `VideoTexture` for rendering the video clip.
- New GameObjects in SampleScene: `Canvas` and `RawImage` for rendering. Currently setup to be a full screen renderer. Will be changed in future.

Code style: [Google ](https://google.github.io/styleguide/csharp-style.html)

### CODE REVIEW

File: Assets/Scripts/Input/InputController.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using UnityEngine.Video;
+
+[RequireComponent(typeof(VideoPlayer))]
+public class InputController : MonoBehaviour {
+  public enum InputMode {
+    Video,
+    Webcam,
+  };
+
+  [SerializeField] private InputMode _inputMode;
```
Comment (wixette) [inline Assets/Scripts/Input/InputController.cs:28]: Why make this kind of serializable field private, instead of a simple `public InputMode InputMode` without the `[SerializeField]` attribute?

A field that can be set in the Unity editor implies that it is accessible publicly in the editor. We might need some reasons to give a field public accessibility in the Unity world while keeping it private in the C# world.

Comment (henryouly) [reply Assets/Scripts/Input/InputController.cs:28]: There are some discussion on Unity forum when I research this topic:
https://forum.unity.com/threads/reason-s-to-serializefield-a-private-variable.513806/

A short answer may be just adopting this practice from other recommendation, however I'm not sure if I'm following the best practice or simply added complexity.

TL;DR I'm trying to separate two things:
- Whether show this field to Inspector
- Whether to allow public access to change the variable (from other scripts)

I'm particular sold on this comment, although not running into it yet: "The first time you encounter a bug where you wrote a class never expecting a variable to be modified directly outside of the class, but you accidentally do so anyway, is where you will understand why to use "public" sparingly. But you may still want to customize via the inspector, so SerializeField is handy."

Do you think it make sense?

Comment (wixette) [reply Assets/Scripts/Input/InputController.cs:28]: Hmm... interesting... I personally treat it as a style-like discussion.

* Public accessibility makes things simple and consistent (between editor's inspector and the C# world).
* A private field with `[SerializeField]` prevents over-access from C# code.

pros and cons as usual. Okay, I understand your reasons now. We can make this usage a style choice at least in this project/repo if you prefer it.

**Review by wixette — APPROVED** [review]

File: Assets/Scripts/Input/InputController.cs (hunk 1 above)
Comment (henryouly) [reply Assets/Scripts/Input/InputController.cs:28]: Cool. I lean towards keeping it then. Thanks for pointing it out.
